### ISSUE TITLE
FINAL field conversion does not consider EML MODIFY statements with "SET FIELDS WITH" as write access

### ISSUE THREAD

**Issue by b4loghpeter** [SAP/abap-cleaner#117]
Dear Community,

The 'Use FINAL for immutable variables' formatting option does not consider some EML statements as write access. It therefore transforms the inline DATA() declarations to FINAL() which leads to syntax errors.

```
 get_rap_create_data(
     IMPORTING et_rap_create_root  = DATA(lt_create_root_data)      "<< Replaced with FINAL
               et_rap_create_child = DATA(lt_create_child_data) ).  "<< Replaced with FINAL

  MODIFY ENTITIES OF <CDS view>
         ENTITY <root_entity>
         CREATE FROM lt_create_root_data      "<< This is OK

         ENTITY <child_entity>
         CREATE BY \_<association>
         SET FIELDS WITH lt_create_child_data "<< Syntax error: "The field lt_create_child_data cannot be modified"

         FAILED   failed_early
         REPORTED reported_early
         MAPPED   mapped.
```

Can you please check whether this can be fixed?

Thanks & Regards,
Peter

**Comment by jmgrassau** [comment]
Hi Peter, 

thanks a lot for reporting this – this particular write position was indeed not yet considered by ABAP cleaner! 

The issue should be fixed with the next release. Unfortunately, with the different variants of ```FROM fields_tab``` / ```AUTO FILL CID WITH fields_tab``` / ```FIELDS ( comp1 comp2 ... ) WITH fields_tab``` / ```[AUTO FILL CID] SET FIELDS WITH fields_tab```, it seems that it is not always possible to statically determine read-only cases. Therefore, ABAP cleaner now _always_ assumes a write access (shown in red) to these fields tables …

![image](https://github.com/SAP/abap-cleaner/assets/79639298/11a80012-ed81-4a48-8cb9-1c2acccf956e)

… to avoid introducing FINAL(…) where it is wrong. 

Kind regards,
Jörg-Michael

**Comment by jmgrassau** [comment]
Hi Peter, 

thanks again for reporting this! It should now be fixed with version 1.6.0, which was just released.

Kind regards,
Jörg-Michael
